direnv.el

```diff
@@ -230,6 +262,25 @@ Same as ‘provided-mode-derived-p’ which is Emacs 26.1+ only."
 (when (fboundp 'provided-mode-derived-p)
   (defalias 'direnv--provided-mode-derived-p 'provided-mode-derived-p))
 
+(defun direnv--callback (items show-summary old-directory)
+  (let ((summary (direnv--summarise-changes items)))
+    (when (and direnv-always-show-summary (not (string-empty-p summary)))
+      (setq show-summary t))
+    (when show-summary
+      (direnv--show-summary summary old-directory direnv--active-directory))
+    (dolist (pair items)
+      (let ((name (car pair))
+            (value (cdr pair)))
+        (setenv name value)
+        (when (string-equal name "PATH")
+          (setq exec-path (append (parse-colon-path value) (list exec-directory)))
+          ;; Prevent `eshell-path-env` getting out-of-sync with $PATH:
+          ;; eshell-path-env is deprecated in newer versions of Emacs
+          (when (derived-mode-p 'eshell-mode)
```

Comment:
This is sensitive to the buffer that it runs in. I don't see any code to manage which buffer the callback runs in, so I expect this callback is getting run in the ` *direnv stdout*` buffer. It works out fine for the `setenv` and `exec-path` bits (as long as `process-environment` and `exec-path` haven't been made buffer local there), but this eshell path maintenance won't work.